crates/runtime/src/instance.rs

```diff
@@ -95,6 +94,52 @@ pub(crate) struct Instance {
     /// index of the slot in the pooling allocator.
     index: usize,
 
+    /// A pointer to the `vmctx` field at the end of the `Instance`.
+    ///
+    /// If you're looking at this a reasonable question would be "why do we need
+    /// a pointer to ourselves?" because after all the pointer's valid is
+    /// trivially derivable from any `&Instance` pointer. The rationale for this
```

Comment:
I'm not sure how to parse "the pointer's valid is trivially derivable from any `&Instance` pointer.

Reply:
Oops I typo'd and it's supposed to be `s/valid/value/`